File: internal/model/public_directory.go
package model

import "time"

// DiskPublicDirectory 公共目录映射表
type DiskPublicDirectory struct {
	ID          uint64    `gorm:"primaryKey;autoIncrement" json:"id"`
	FolderID    uint64    `gorm:"uniqueIndex;not null" json:"folderId"`
	Scope       string    `gorm:"size:16;not null" json:"scope"`
	Department  string    `gorm:"size:64;not null;default:''" json:"department"`
	DisplayName string    `gorm:"size:255;not null" json:"displayName"`
	FixedPath   string    `gorm:"size:512;not null;default:''" json:"fixedPath"`
	CreatedBy   string    `gorm:"size:64;not null;default:''" json:"createdBy"`
	IsActive    bool      `gorm:"default:true" json:"isActive"`
	CreatedAt   time.Time `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt   time.Time `gorm:"autoUpdateTime" json:"updatedAt"`
}

func (DiskPublicDirectory) TableName() string { return "disk_public_directory" }

File: sdk/agentdisk/models/public_directory.py
"""Public directory model."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass
class DiskPublicDirectory:
    """A public directory visible to authenticated users."""

    id: int
    folder_id: int
    scope: str
    department: str
    display_name: str
    fixed_path: str
    is_active: bool
    created_by: str


def from_dict(data: dict) -> DiskPublicDirectory:
    return DiskPublicDirectory(
        id=data.get("id", 0),
        folder_id=data.get("folderId", 0),
        scope=data.get("scope", ""),
        department=data.get("department", ""),
        display_name=data.get("displayName", ""),
        fixed_path=data.get("fixedPath", ""),
        is_active=data.get("isActive", True),
        created_by=data.get("createdBy", ""),
    )
